utils/bootfile/editor.go

```diff
@@ -0,0 +1,121 @@
+package bootfile
+
+import (
+	"bufio"
+	"os"
+	"strings"
+
+	logging "github.com/home-assistant/os-agent/utils/log"
+)
+
+func ReadOption(filePath string, optionName string, defaultValue string) (string, error) {
+	// Read the options from the boot file
+	file, err := os.Open(filePath)
+	if err != nil {
+		logging.Error.Printf("Failed to open boot file %s: %s", filePath, err)
+		return defaultValue, err
+	}
+	defer file.Close()
+
+	// Scan over all lines
+	fileScanner := bufio.NewScanner(file)
+	fileScanner.Split(bufio.ScanLines)
+
+	for fileScanner.Scan() {
+		line := fileScanner.Text()
+		if strings.HasPrefix(line, optionName) {
+			return strings.Replace(line, optionName, "", 1), nil
+		}
+	}
+
+	return defaultValue, nil
+}
+
+func DisableOption(filePath string, optionName string) error {
+	// Read the options from the boot file
+	file, err := os.Open(filePath)
+	if err != nil {
+		logging.Error.Printf("Failed to open boot file %s: %s", filePath, err)
+		return err
+	}
+
+	// Scan over all lines
+	fileScanner := bufio.NewScanner(file)
+	fileScanner.Split(bufio.ScanLines)
+
+	var outLines []string
+	for fileScanner.Scan() {
+		line := fileScanner.Text()
+		if strings.HasPrefix(line, optionName) {
+			outLines = append(outLines, "#"+line)
+		} else {
+			outLines = append(outLines, line)
+		}
+	}
+	file.Close()
+
+	// Write all lines back to boot config file
+	file, err = os.Create(filePath)
```

Comment:
This does not look atomic. Probably low risk for this usage, but it is a smell. You might want to write a temporary file and `os.Rename` it into place.

Reply:
Yeah good point. Given that its a boot file we should be careful about it. There seem to be Go libraries to write atomic, e.g. https://github.com/natefinch/atomic

Reply:
Addressed with #99.